### ISSUE TITLE
Camera broken

### ISSUE THREAD

**Issue by Pandapip1** [PrismarineJS/prismarine-viewer#42]
The problem is in `src/index.js`. The problem lines are:
```js
camera.rotation.x = camera.rotation.x * 0.9 + pitch * 0.1
camera.rotation.y = camera.rotation.y * 0.9 + yaw * 0.1
```
This only works directionally. The fix is simple:
```js
camera.lookAt(pos.x-Math.cos(pitch)*Math.sin(yaw), pos.y-Math.sin(pitch)+1.6, pos.z-Math.cos(pitch)*Math.cos(yaw))
```

**Comment by Karang** [comment]
Can you develop a bit on what problem it is supposed to solve ?

The code you propose is equivalent to:
```js
camera.rotation.x = pitch
camera.rotation.y = yaw
```
(camera.rotation is an instance of https://threejs.org/docs/#api/en/math/Euler x, y, z are the euler angles pitch, yaw, roll)

The interpolation is there to smooth the motion a bit. I'm not saying there is no problem, but I haven't seen one when testing.

**Comment by Pandapip1** [comment]
For some reason, for me it doesn't work when you are facing the +x or -x direction; it rolls instead.

**Comment by Karang** [comment]
Interesting. Do you have a minimal script that can reproduce the issue ?

**Comment by Pandapip1** [comment]
Not really, but I'll make a pull request with the fix.

**Comment by Pandapip1** [comment]
#43

**Comment by Karang** [comment]
The bug lies in the order the rotation are applied, changing the code to:
```js
camera.rotation.set(pitch, yaw, 0, 'ZYX')
```
fix the issue. Will be merged in #56
